internal/controller/git/git.go

```diff
@@ -758,6 +763,9 @@ func (r *repo) buildCommand(command string, arg ...string) *exec.Cmd {
 
 func (r *repo) buildGitCommand(arg ...string) *exec.Cmd {
 	cmd := r.buildCommand("git", arg...)
+	if r.useSSH {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_SSH_COMMAND=ssh -F %s/.ssh/config", r.homeDir))
+	}
```

Comment:
```suggestion
	cmd.Env = append(
		cmd.Env,
		fmt.Sprintf(
			"GIT_SSH_COMMAND=ssh -F %s/.ssh/config -i %s/.ssh/id_rsa",
			r.homeDir, r.homeDir,
		),
	)
```

@gnadaban did you ever see this working end-to-end? In all of my experimentation, this will fail if the location of the keyfile isn't also specified.

Reply:
I actually came up with the solution by making this work in our environment, so I can confirm my suggested solution works.

Reply:
@krancour , in your suggestion you removed the IdentityFile setting from the SSH config, that is likely the reason.

Was that intentional?

Reply:
Nope. Oversight on my part -- and this explains it. Thanks.

You can ignore this suggestion!